src/test/unit/services/pool.spec.ts

```diff
@@ -0,0 +1,111 @@
+import { GatewayService } from "src/common/gateway/gateway.service";
+import { PoolFilter } from "src/endpoints/pool/entities/pool.filter";
+import { PoolService } from "src/endpoints/pool/pool.service";
+import { TransactionType } from "src/endpoints/transactions/entities/transaction.type";
+
+describe('PoolService', () => {
+  let service: PoolService;
+  let gatewayService: GatewayService;
+  let cacheService: CacheService;
+
+  beforeEach(async () => {
+    const moduleRef = await Test.createTestingModule({
+      providers: [
+        PoolService,
+        {
+          provide: GatewayService,
+          useValue: {
+            getTransactionPool: jest.fn(),
+          },
+        },
+        {
+          provide: CacheService,
+          useValue: {
+            getOrSet: jest.fn(),
+          },
+        },
+        {
+          provide: ApiConfigService,
+          useValue: {
+            isTransactionPoolEnabled: jest.fn().mockResolvedValue(true),
+          },
+        },
+      ],
+    }).compile();
+
+    service = moduleRef.get<PoolService>(PoolService);
+    gatewayService = moduleRef.get<GatewayService>(GatewayService);
+    cacheService = moduleRef.get<CacheService>(CacheService);
+
+    const data = require('../../mocks/pool.mock.json');
+
+    gatewayService.getTransactionPool = jest.fn().mockResolvedValue(data);
+    const txPoolRaw = await service.getTxPoolRaw();
+
+    cacheService.getOrSet = jest.fn().mockResolvedValue(txPoolRaw);
+
+  });
+
+  it('service should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  describe('getPool', () => {
+    it('should work and return the pool', async () => {
+      const pool = await service.getPool(new QueryPagination(), new PoolFilter());
+      expect(pool).toHaveLength(7);
+      expect(pool[0].type).toBe(TransactionType.Transaction);
+    });
+
+    it('should work and return the pool with filters', async () => {
+      let pool = await service.getPool(new QueryPagination(), new PoolFilter({ type: TransactionType.Transaction }));
+      expect(pool).toHaveLength(1);
+
+      pool = await service.getPool(new QueryPagination(), new PoolFilter({ type: TransactionType.SmartContractResult }));
+      expect(pool).toHaveLength(1);
+
+      pool = await service.getPool(new QueryPagination(), new PoolFilter({ type: TransactionType.Reward }));
+      expect(pool).toHaveLength(5);
+    });
+
+    it('should work and return the pool with query pagination', async () => {
+      const pool = await service.getPool(new QueryPagination({ from: 0, size: 2 }), new PoolFilter({ type: TransactionType.Reward }));
+      expect(pool).toHaveLength(2);
+      expect(pool[0].type).toBe(TransactionType.Reward);
+    });
+  });
+
+  describe('getPoolCount', () => {
+    it('should work and return the pool count', async () => {
+      const poolCount = await service.getPoolCount(new PoolFilter());
+      expect(poolCount).toBe(7);
+    });
```

Comment:
Instead of using `.toBe` use `.toStrictEqual`

Reply:
done